GavinJoyce/ember-headlessui
# ✨ Implement `<RadioGroup />` component

## ✨ What Changed and Why
This PR is a functionally complete implementation of a `<RadioGroup />` component based on [headlessui.dev/react/radio-group](https://headlessui.dev/react/radio-group)

Things that this PR implements:

- Support for disabled component and individual options
- Support for a optional label element
- Full mouse support for focus and active elements
- Full keyboard support (Enter/Space/ArrowUp/ArrowDown/ArrowLeft/ArrowRight)
- Full accessibility implementation and focus management
- Sample implementation in dummy app
- Ported the 30 [React component tests for radio-group](https://github.com/tailwindlabs/headlessui/blob/6d8235e059d492f67c6a3ebe455b5da658f0bf2e/packages/%40headlessui-react/src/components/radio-group/radio-group.test.tsx) as Ember integration tests. 

## 😈 Before
NA

## 😇 After

https://user-images.githubusercontent.com/43208685/166691121-80f87c55-d1ae-469f-b7b8-92255a0197a7.mp4

## 🌈 Testing
1. Check out to branch `phorest:feature/radio-group`
1. Visit http://localhost:4200/radio-group

Closes: https://github.com/GavinJoyce/ember-headlessui/issues/103
